src/hipscat_import/arguments.py

```diff
@@ -115,44 +114,52 @@ def _check_paths(self):
         os.makedirs(self.catalog_path, exist_ok=True)
 
         # Basic checks complete - make more checks and create directories where necessary
-        if self.input_path:
-            if not os.path.exists(self.input_path):
+        if self._input_path:
+            if not os.path.exists(self._input_path):
                 raise FileNotFoundError("input_path not found on local storage")
-            self.input_paths = glob.glob(f"{self.input_path}/*{self.input_format}")
+            self.input_paths = glob.glob(f"{self._input_path}/*{self.input_format}")
             if len(self.input_paths) == 0:
                 raise FileNotFoundError(
-                    f"No files matched file pattern: {self.input_path}*{self.input_format} "
+                    f"No files matched file pattern: {self._input_path}*{self.input_format} "
                 )
-        elif self.input_file_list:
-            self.input_paths = self.input_file_list
+        elif self._input_file_list:
+            self.input_paths = self._input_file_list
             for test_path in self.input_paths:
                 if not os.path.exists(test_path):
                     raise FileNotFoundError(f"{test_path} not found on local storage")
+        if not self.input_paths:
+            raise FileNotFoundError("No input files found")
+        self.input_paths.sort()
 
-        # Create a temp folder unique to this execution of the partitioner.
-        # This avoids clobbering other executions.
-        tmp_prefix = "/tmp"
-        if self.tmp_dir:
-            tmp_prefix = self.tmp_dir
+        if self._tmp_dir:
+            self.tmp_path = os.path.join(
+                self._tmp_dir, self.catalog_name, "intermediate"
+            )
         elif self.dask_tmp:
-            tmp_prefix = self.dask_tmp
+            self.tmp_path = os.path.join(
+                self.dask_tmp, self.catalog_name, "intermediate"
+            )
         else:
-            tmp_prefix = self.output_path
-        # pylint: disable=consider-using-with
-        tmp_dir = tempfile.TemporaryDirectory(dir=tmp_prefix)
-        self.tmp_dir = tmp_dir.name
-        self.contexts.append(tmp_dir)
+            self.tmp_path = os.path.join(self.catalog_path, "intermediate")
+        if not self.resume:
+            existing_tmp_files = glob.glob(f"{self.tmp_path}/*")
+            if existing_tmp_files:
+                raise ValueError(
+                    f"tmp_path ({self.tmp_path}) contains intermediate files."
+                    " choose a different directory or use --resume flag"
+                )
+        os.makedirs(self.tmp_path, exist_ok=True)
```

Comment:
I want to make sure that I understand this change of behavior. Moving from the `tempfile.TemporaryDirectory` to `os.makedirs` means the temporary file directory persists if the job is killed. The advantage of that the intermediate files exist for a resume. The disadvantage is that you now need to clean up. Is that all correct?